src/_routing/_components/utils.ts

```diff
@@ -0,0 +1,14 @@
+import { TMainRoutes } from './MainNavigator';
+
+export function mapRouteNameToIcon(route: TMainRoutes): 'Profile' | 'QR' | 'Shop' {
+  switch (route) {
+    case 'Profile':
+      return 'Profile';
+    case 'Scan':
+      return 'QR';
+    case 'Shop':
+      return 'Shop';
+    default:
+      break;
+  }
```

Comment:
Would be more typesafe if we'd just use a key-value object typed as `Record<TMainRoutes, TIconName>`